### PR TITLE
upgrade maven-shade-plugin version to 3.5.2

### PR DESCRIPTION
**Pull request by xiangfu0**
Upgrade maven-shade-plugin version from 3.2.x to 3.5.2.
By default, modules are not building the shaded jar.
For the modules require shaded jars, users can specify `<shade.phase.prop>package</shade.phase.prop>` in the `pom.xml` file `properties` section.

### CODE REVIEW

File: .github/workflows/scripts/pr-tests/.pinot_tests_unit.sh [hunk 1]
```diff
@@ -33,6 +33,7 @@ if [ "$RUN_TEST_SET" == "1" ]; then
       -pl 'pinot-spi' \
       -pl 'pinot-segment-spi' \
       -pl 'pinot-common' \
+      -pl ':pinot-yammer' \
```
Comment (Jackie-Jiang) [inline .github/workflows/scripts/pr-tests/.pinot_tests_unit.sh:36]: Is this related?

Comment (xiangfu0) [reply .github/workflows/scripts/pr-tests/.pinot_tests_unit.sh:36]: yes, the new shade plugin doesn't build the test jar, so need to explicitly specify this here.

Comment (Jackie-Jiang) [reply .github/workflows/scripts/pr-tests/.pinot_tests_unit.sh:36]: I don't fully follow. Does the first test set depend on `pinot-yammer`? If so, should we exclude it from the second test set?
What does the leading `:` stand for?

Comment (xiangfu0) [reply .github/workflows/scripts/pr-tests/.pinot_tests_unit.sh:36]: yes, `pinot-common` uses `pinot-yammer` as test-jar dependency
`pinot-yammer` is under a path like `pinot-plugins/pinot-metrics/pinot-yammer`, so using `:pinot-yammer` will match the module path.

Comment (Jackie-Jiang) [reply .github/workflows/scripts/pr-tests/.pinot_tests_unit.sh:36]: In that case, should we move the `pinot-metrics` tests into this test set and exclude it from the second test set?

Comment (xiangfu0) [reply .github/workflows/scripts/pr-tests/.pinot_tests_unit.sh:36]: It's a bit complicated as we need to put all the submodules of `pinot-metrics` into .
Also this `pinot-yammer` is a very small test, so it's ok to run it double times.

File: pinot-perf/pom.xml [hunk 2]
```diff
@@ -214,7 +214,6 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-shade-plugin</artifactId>
-        <version>3.1.0</version>
```
Comment (Jackie-Jiang) [inline pinot-perf/pom.xml:217]: Do we need to keep this plugin? I guess we should never package pinot-perf?

File: pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-common/pom.xml [hunk 3]
```diff
@@ -35,6 +35,6 @@
   <url>https://pinot.apache.org/</url>
   <properties>
     <pinot.root>${basedir}/../../..</pinot.root>
-    <phase.prop>none</phase.prop>
+    <shade.phase.prop>none</shade.phase.prop>
```
Comment (Jackie-Jiang) [inline pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-common/pom.xml:38]: (minor) Not needed?

File: pinot-plugins/pinot-file-system/pinot-s3/pom.xml [hunk 4]
```diff
@@ -65,55 +65,4 @@
       <scope>test</scope>
     </dependency>
   </dependencies>
-
-  <profiles>
-    <profile>
-      <id>build-shaded-jar</id>
-      <activation>
-        <property>
-          <name>skipShade</name>
```
Comment (Jackie-Jiang) [inline pinot-plugins/pinot-file-system/pinot-s3/pom.xml:74]: Do we want to keep this option in the main pom?

File: pom.xml [hunk 5]
```diff
@@ -2209,6 +2211,9 @@
             <!-- Exclude build targets -->
             <exclude>**/target/**</exclude>
 
+            <!-- Exclude Maven plugin generated files -->
+            <exclude>**/dependency-reduced-pom.xml</exclude>
```
Comment (Jackie-Jiang) [inline pom.xml:2215]: I never saw this file generated

**Review by Jackie-Jiang — APPROVED** [review]
Nice!

File: pinot-plugins/pinot-file-system/pinot-s3/pom.xml (hunk 4 above)
Comment (xiangfu0) [reply pinot-plugins/pinot-file-system/pinot-s3/pom.xml:74]: I don't see the usage for this for a while.

File: pinot-perf/pom.xml (hunk 2 above)
Comment (xiangfu0) [reply pinot-perf/pom.xml:217]: this is not packaged to pinot-distribution, it's trying to generate another main class for perf testing.

File: pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-common/pom.xml (hunk 3 above)
Comment (xiangfu0) [reply pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-common/pom.xml:38]: true

File: pom.xml (hunk 5 above)
Comment (xiangfu0) [reply pom.xml:2215]: new version plugin generates  this file

File: pinot-spi/pom.xml [hunk 6]
```diff
@@ -179,39 +179,4 @@
       <artifactId>reflections</artifactId>
     </dependency>
   </dependencies>
-  <profiles>
-    <profile>
-      <id>build-shaded-jar</id>
-      <activation>
-        <activeByDefault>false</activeByDefault>
-      </activation>
```
Comment (jasperjiaguo) [inline pinot-spi/pom.xml:187]: After removing this these module level profile the mvn build stopped to publish shaded jars for spi which is required by our build pipeline. We should probably add it back.
